my-project/src/Controller/ProductCategoryController.php

```diff
@@ -34,7 +35,29 @@ class ProductCategoryController extends Controller
      */
     public function index(ProductCategoryRepository $productCategoryRepository): Response
     {
-        return $this->render('product_category/index.html.twig', ['product_categories' => $productCategoryRepository->findAll()]);
+        $AllCategories = $productCategoryRepository->findAll();
+        $filesystem = new Filesystem();
+        $imagesFolderPath = "uploads/images/";
+        /** @var ProductCategory $category */
+        foreach ($AllCategories as $category)
+        {
+            /** @var string $mainImagePath */
+            $mainImagePath = $imagesFolderPath.$category->getMainImage();
```

Comment:
@mSobonVirtua wont this find an media folder when no filename is set int the database?

Reply:
@astyczen Filename is always set in database. When you create new ProductCategory you must set mainImage. You cannot delete it, you can only change it to another.